WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationCommentDetailViewController.swift

```diff
@@ -156,21 +164,41 @@ private extension NotificationCommentDetailViewController {
             return
         }
 
-        if commentDetailViewController != nil {
-            commentDetailViewController?.refreshView(comment: comment, notification: notification)
+        // Refresh the current content if the underlying view controller supports it
+        // Else, remove the existing child view controller and add a new one.
+        if let commentDetailViewController {
+            commentDetailViewController.refreshView(comment: comment, notification: notification)
         } else {
-            let commentDetailViewController = CommentDetailViewController(comment: comment,
-                                                                      notification: notification,
-                                                                      notificationDelegate: notificationDelegate,
-                                                                      managedObjectContext: managedObjectContext)
-
-            commentDetailViewController.view.translatesAutoresizingMaskIntoConstraints = false
-            add(commentDetailViewController)
-            view.pinSubviewToAllEdges(commentDetailViewController.view)
-            self.commentDetailViewController = commentDetailViewController
+            self.content?.viewController.remove()
+            let newContent = makeNewContent(with: comment)
+            let viewController = newContent.viewController
+            viewController.view.translatesAutoresizingMaskIntoConstraints = false
+            self.add(viewController)
+            self.view.pinSubviewToAllEdges(viewController.view)
+            self.content = newContent
         }
 
-        configureNavBarButtons()
+        self.configureNavBarButtons()
+    }
+
+    private func makeNewContent(with comment: Comment) -> Content {
+        if let blog = comment.blog,
```

Comment:
This If statement seems a bit complex and hard to follow. Is there a way we can simplify it?

Reply:
I feel the same, I'll see how to improve readability. 🧘

Reply:
Resolved in 7b0677405d7e8daad522222ee36ca6e18cdeb44b